doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -232,42 +397,486 @@ that common prefixes use the samel levels.
 
 .. coqtop:: all
 
-   Reserved Notation "x << y" (at level 70).
-   Fail Reserved Notation "x << y << z" (at level 70, y at level 200).
+   Reserved Notation "x << y" (at level 70).
+   Fail Reserved Notation "x << y << z" (at level 70, y at level 200).
+
+In order to factorize the left part of the rules, the subexpression
+referred to by ``y`` has to be at the same level in both rules. However the
+default behavior puts ``y`` at the next level below 70 in the first rule
+(``no associativity`` is the default). To fix this, we
+need to force the parsing level of ``y``, as follows.
+
+.. coqtop:: reset all
+
+   Reserved Notation "x << y" (at level 70).
+   Reserved Notation "x << y << z" (at level 70, y at next level).
+
+Or better yet, simply let the defaults ensure the best factorization.
+
+.. coqtop:: reset all
+
+   Reserved Notation "x << y" (at level 70).
+   Reserved Notation "x << y << z".
+   Print Notation "_ << _ << _".
+
+For the sake of factorization with Coq predefined rules, simple rules
+have to be observed for notations starting with a symbol, e.g., rules
+starting with “\ ``{``\ ” or “\ ``(``\ ” should be put at level 0. The list
+of Coq predefined notations can be found in the chapter on :ref:`thecoqlibrary`.
+
+.. warn:: Closed notations (i.e. starting and ending with a terminal symbol) should usually be at level 0 (default).
+   :name: closed-notation-not-level-0
+
+   It is usually better to put closed notations, that is the ones starting and ending with a terminal symbol, at level 0.
```

Comment:
Why is that so? What does this "usually" hides?

Reply:
Here usually means almost always, except in a few very specific cases that I don't precisely remember.